src/commands/login.rs

```diff
@@ -10,9 +13,57 @@ pub struct LoginSubcommand {
     pub token: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct DeviceCodeResponse {
+    device_code: String,
+    user_code: String,
+    verification_uri: String,
+    expires_in: u64,
+    interval: u64,
+}
+
+#[derive(Deserialize)]
+struct DeviceCodeAuth {
+    access_token: String,
+    token_type: String,
+    scope: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum AuthResponse {
+    Ok(DeviceCodeAuth),
+    Err { error: String },
+}
+
+fn await_github_auth(device_code_response: DeviceCodeResponse) -> anyhow::Result<DeviceCodeAuth> {
+    let client = reqwest::blocking::Client::new();
+
+    thread::sleep(time::Duration::from_secs(device_code_response.interval));
+
+    let response = client
+        .post("https://github.com/login/oauth/access_token")
+        .header("accept", "application/json")
+        .json(&serde_json::json!({
+            "client_id": "7bd503594a0f9a9f7ed3",
```

Comment:
Let's make this a const in the file!

Reply:
This now gets configured in the registry index! If you define `oauth_id` in the index with this then clients will use it to attempt to use oauth to sign in. This way it isn't just us who can define oauth targets.